Chug not assigned

JonahZW opened the pull request:
## Work Done
### Missing Chug Assignment(s) Report
- Generates report for campers who have not been assigned a chug (can filter by block, edah, and chug group)
- Some formatting on report menu (creating consistent styling of popup guides)
- Opt-out of generating checkboxes for certain reports (for both the HTML version and exported ones) by passing an additional parameter through (and adjusting the widths of each column row so everything is more visible) -- went in and removed checkboxes from reports in which I found it unnecessary

### Don't Assign Campers to Chug if no Preferences
- See title
- Designate campers without preferences on levelHome
    - Highlight campers with no preferences in purple
    - Improve instructions to include this information

## Testing Instructions
After running `./start.sh` to start the docker images, run `docker ps` and get the container id for image `mysql`

Using that container id, run (if id is 3342c74693b9) `docker exec -it 3342c74693b9 mysql -uroot -pdeveloper`

Copy and paste the following SQL commands (you may need to press enter after):
```
use camprama_chugbot_db;
insert into campers values(3,1,1,NULL,'Johnny','Appleseed','dlobron@gmail.com',0,0),(4,1,1,NULL,'David','Offit','dlobron@gmail.com',0,0); -- add additional campers
insert into blocks values(2,'Weeks 3+4',1); -- add additional chug block
insert into block_instances values(2,1); -- make block available for first session
insert into edot_for_block values(2,1); -- make block available for kochavim
insert into chug_groups values(2,'Bet'); -- add additional chug group
insert into edot_for_group values(2,1); -- make chug group available for kochavim
insert into chugim values('Dance',2,10000,-1,NULL,8),('Rikud',2,10000,-1,NULL,9),('Shira',2,10000,-1,NULL,10),('Zimkudiyah',2,10000,-1,NULL,11),('Shtik',2,10000,-1,NULL,12),('Clowning',2,10000,-1,NULL,13),('Juggling',2,10000,-1,NULL,14); -- add chugim for chug bet
insert into edot_for_chug values(8,1),(9,1),(10,1),(11,1),(12,1),(13,1),(14,1); -- make chugim available for kochavim
insert into chug_instances values(8,2,8),(9,2,9),(10,2,10),(11,2,11),(12,2,12),(13,2,13),(14,2,14),(8,1,15),(9,1,16),(10,1,17),(11,1,18),(12,1,19),(13,1,20),(14,1,21),(1,2,22),(2,2,23),(3,2,24),(4,2,25),(5,2,26),(6,2,27),(7,2,28); -- offer chugim for correct prakim for each time block
insert into preferences values(3,1,1,4,6,2,1,5,7,3),(2,2,1,9,8,10,12,11,14,4),(4,2,1,14,13,12,11,10,9,5); -- add some preferences for first block (weeks 1+2)
insert into preferences values(1,1,2,1,7,5,6,3,2,6),(2,1,2,7,6,5,4,3,2,7),(3,1,2,1,3,5,2,4,6,8),(4,2,2,14,12,10,13,11,9,9); -- add preferences for second block (weeks 3+4) (note - 3 campers now have preferences for chug aleph, but only 1 for chug bet)
```

Log into staff home (pwd: developer) and run the leveling report (you may need to click reassign, some matches already are made in the database and we’re building on those).
Expected output:
- Chug aleph, weeks 1+2: all assigned except David
- Chug bet, weeks 1+2: David and Johannes assigned, Igor and Johnny unassigned
- Chug aleph, weeks 3+4: all assigned except David
- Chug bet, weeks 3+4: all unassigned except David
    - Anyone unassigned should be in the “Not Assigned Yet” category, and their color should be purple, even if moved to a different category

Go to reports, and run the report for “Campers Not Assigned to Chug” with no additional options. The complete results from above will be reflected in a table.

Feel free to continue to experiment with assigning campers to chugim and adding/combining filters to the report. 

Additional note: when I was working locally, changes in the JS file did not apply by default because the file had been cached. You may need to clear your cache (and/or while on the local version of the website, go to Inspect Element-> Network->Disable caching) to see the changes.

JonahZW committed:
Missing Chug Assignment(s) Report

- Generates report for campers who have not been assigned a chug (can filter by block, edah, and chug group)
- Some formatting on report menu (creating consistent styling of popup guides)
- Opt-out of generating checkboxes for certain reports (for both the HTML version and exported ones) by passing an additional parameter through (and adjusting the widths of each column row so everything is more visible) -- went in and removed checkboxes from reports in which I found it unnecessary

JonahZW committed:
Don't assign campers without preferences

JonahZW committed:
Designate campers without preferences on levelHome

- Highlight campers with no preferences in purple
- Improve instructions to include this information

JonahZW committed:
Handle when campers have some preferences but not others

dlobron reviewed (approved):
This looks great! I've tested it in Docker, and it looks good so far! 

One minor suggestion: if you could add a comment on line 257 of assignment.php, just noting that if $row[4] is null, it means that there are no preferences and the camper will not be included in the leveling.

JonahZW committed:
add comment

note in comments of assignment.php where campers are passed over and not assigned to a chug if missing preferences for that perek

JonahZW committed:
Update ChugBotWithData.sql

update admin_data to include enable_camper_importer and enable_selection_process

JonahZW commented:
I also just made a small change to ChugBotWithData.sql, updating it to include fields in `admin_data` for `enable_camper_importer` and `enable_selection_process`, two fields which are expected to exist in the admin settings page, but which were not previously present.

dlobron committed:
Add missing fields to other .sql file.

dlobron reviewed (approved):
Looks great, Jonah! I copied your change over to the other .sql file as well.
